api/v1alpha2/cloudprovider_types.go

```diff
@@ -234,3 +242,26 @@ type CPILabelConfig struct {
 	// +optional
 	Region string `gcfg:"region,omitempty" json:"region,omitempty"`
 }
+
+// CPIRouteConfig is the vSphere cloud provider's route configuration.
+type CPIRouteConfig struct {
+	// T1RouterID is NSX-T T1 logical router ID used to create static route on Tier1 gateway.
+	// +optional
+	T1RouterID string `gcfg:"t1-router-id,omitempty" json:"t1-router-id,omitempty"`
+}
+
+// CPINSXTConfig is the vSphere cloud provider's NSX-T configuration.
+type CPINSXTConfig struct {
+	// SecretName is the name of the Kubernetes secret in which the NSX-T
+	// credentials are located.
+	// +optional
+	SecretName string `gcfg:"secret-name,omitempty" json:"secretName,omitempty"`
+
+	// SecretNamespace is the namespace for SecretName.
+	// +optional
+	SecretNamespace string `gcfg:"secret-namespace,omitempty" json:"secretNamespace,omitempty"`
+
+	// Server is the NSX-T server.
+	// +optional
+	Server string `gcfg:"server,omitempty" json:"server,omitempty"`
```

Comment:
Unfortunately, I think we need to keep this as `host` for now, see this thread https://github.com/kubernetes/cloud-provider-vsphere/pull/407#discussion_r545356907

Reply:
I see. Will update